dipy/align/_public.py

```diff
@@ -404,6 +407,14 @@ def affine_registration(moving, static,
         Set it to True to return the value of the optimized coefficients and
         the optimization quality metric.
 
+    moving_mask : array, shape (S', R', C') or (R', C'), optional
```

Comment:
Do you want to enable passing array nifti or str like for the data? This might also allow passing masks that are in a different space than the data, which could be useful (for example, a brain mask from a T1w image)

Reply:
It's an interesting idea, are you suggesting that this would be useful for aligning images to a T1w image that had a mask already defined, and saved somewhere?

In this case, can't the user just load the mask and pass in the array?

In any case, the public API needs updating as it only accepts 3D data (technically 4D i.e. a series of 3D volumes). I will begin making these modifications later and will consider this change at that stage.